phpunit/fonts-library/class-wp-font-family-utils-test.php

```diff
@@ -0,0 +1,132 @@
+		);
+		$this->assertTrue(
+			WP_Font_Family_Utils::has_font_mime_type( '/temp/piazzolla_400_italic.woff2' )
+		);
+	}
+
+	function test_get_filename_from_font_face() {
+		$font_face = array(
+			'fontFamily' => 'Piazzolla',
+			'fontStyle'  => 'italic',
+			'fontWeight' => '400',
+			'src'        => array(
+				'http://example.com/fonts/piazzolla1.ttf',
+				'http://example.com/fonts/piazzolla2.ttf',
+				'http://example.com/fonts/piazzolla3.ttf',
+			),
+		);
+
+		$this->assertEquals(
+			WP_Font_Family_Utils::get_filename_from_font_face( $font_face, $font_face['src'][0] ),
+			'piazzolla_italic_400.ttf'
+		);
+
+		$this->assertEquals(
+			WP_Font_Family_Utils::get_filename_from_font_face( $font_face, $font_face['src'][1], 2 ),
+			'piazzolla_italic_400_2.ttf'
+		);
+
+		$this->assertEquals(
+			WP_Font_Family_Utils::get_filename_from_font_face( $font_face, $font_face['src'][2], 3 ),
+			'piazzolla_italic_400_3.ttf'
+		);
+	}
+
+	function test_merge_fonts_data() {
+		$font1 = array(
+			'slug'       => 'Piazzolla',
+			'name'       => 'Piazzolla',
+			'fontFamily' => 'Piazzolla',
+			'fontFace'   => array(
+				array(
+					'fontFamily' => 'Piazzolla',
+					'fontStyle'  => 'italic',
+					'fontWeight' => '400',
+					'src'        => 'http://example.com/fonts/piazzolla_400_italic.ttf',
+				),
+				array(
+					'fontFamily' => 'Piazzolla',
+					'fontStyle'  => 'italic',
+					'fontWeight' => '500',
+					'src'        => 'http://example.com/fonts/piazzolla_500_italic.ttf',
+				),
+			),
+		);
+
+		$font2 = array(
+			'slug'       => 'Piazzolla',
+			'fontFamily' => 'Piazzolla',
+			'fontFace'   => array(
+				array(
+					'fontFamily' => 'Piazzolla',
+					'fontStyle'  => 'italic',
+					'fontWeight' => '400',
+					'src'        => 'http://example.com/fonts/piazzolla_400_italic.ttf',
+				),
+				array(
+					'fontFamily' => 'Piazzolla',
+					'fontStyle'  => 'normal',
+					'fontWeight' => '400',
+					'src'        => 'http://example.com/fonts/piazzolla_400_normal.ttf',
+				),
+				array(
+					'fontFamily' => 'Piazzolla',
+					'fontStyle'  => 'italic',
+					'fontWeight' => '700',
+					'src'        => 'http://example.com/fonts/piazzolla_700_italic.ttf',
+				),
+			),
+		);
+
+		$font3 = array(
+			'slug'       => 'encode-sans',
+			'name'       => 'Encode Sans',
+			'fontFamily' => 'Encode Sans',
+		);
+
+		// Fonts with same slug should be merged
+		$merged_font = WP_Font_Family_Utils::merge_fonts_data( $font1, $font2 );
+		$this->assertFalse( $merged_font instanceof WP_Error );
+
+		// Total of font faces without duplicates
+		$this->assertEquals(
+			count( $merged_font['fontFace'] ),
+			4
+		);
+
+		// Missing keys should be added to the merged result
+		$this->assertEquals(
+			$merged_font['name'],
+			'Piazzolla'
+		);
+
+		// Fonts with different slugs should not be merged
+		$merged_font = WP_Font_Family_Utils::merge_fonts_data( $font1, $font3 );
+		$this->assertTrue( $merged_font instanceof WP_Error );
```

Comment:
The `assertWPError()` assertion can be used here.

Reply:
done